### PR TITLE
#162694165 Fix security users form

### PR DESCRIPTION
**Pull request by Arthur236**
## What does this PR do?

Fixes the security users form

## Description of Task to be completed?

```
The modal should conform to the standard forms used throughout the dashboard.
```

## How should this be manually tested?

* Log in
* Click on the `Users` link on the navigation bar
* Click on the `Security Users` tab
* Click on the `ADD SECURITY USER` button to see the modal

## What are the relevant pivotal tracker stories?

[162694165](https://www.pivotaltracker.com/story/show/162694165)

## Any background context you want to add?

N/A

## Important notes

N/A

## Packages installed

N/A

## Deployment note

N/A

## Related PRs branch | PR (branch_name) | (pr_link)

N/A

## Todos
- [x] Raise PR
- [ ] Test
- [ ] Documentation

## Screenshots

When the email is not valid before submitting data to the backend
screenshot 2018-12-20 at 12 53 27

When there is an error after submitting data to the backend
screenshot 2018-12-20 at 13 17 07

When data is submitted successfully without any errors
screenshot 2018-12-20 at 13 17 59

### CODE REVIEW

File: src/_reducers/securityUsers.reducer.js [hunk 1]
```diff
@@ -12,6 +12,22 @@ const {
   RESET_STATUS_MESSAGE
 } = constants;
 
+const getErrorMessage = (error) => {
+  if (error.hasOwnProperty('email')) {
+    return error.email[0].replace(/[[\]']/g, '');
```
Comment (sachungo) [inline src/_reducers/securityUsers.reducer.js]: I take it that the backend team never fixed this weird error message. You can add it as a todo to nag them next year...2019 new year goal 🤣

File: src/components/SecurityUser/AddSecurityUserComponent.jsx [hunk 2]
```diff
@@ -1,78 +1,150 @@
...
+    const { value, name } = event.target;
+    this.setState({
+      [name]: value
+    });
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+
+    if (!emailValidation(this.state.email)) {
```
Comment (sachungo) [inline src/components/SecurityUser/AddSecurityUserComponent.jsx]: Hmmm...this `if...else` is bugging me. Let's see...how about using the following:
```jsx
handleSubmit = (event) => {
  event.preventDefault();
  const isValidEmail = emailValidation(this.state.email);
  if (!isValidEmail) {
    return this.setState({ emailError: !isValidEmail });
  }

  this.setState({ emailError: isValidEmail });

  return this.props.addSecurityUser({
    first_name: this.state.firstName,
    last_name: this.state.lastName,
    phone_number: this.state.phoneNumber,
    email: this.state.email,
    badge_number: this.state.badgeNumber
  });
};
```
Regex [test](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test#Returns) method returns a boolean. Therefore, we can use it's value directly

File: src/_reducers/securityUsers.reducer.js (hunk 1 above)
Comment (Arthur236) [reply src/_reducers/securityUsers.reducer.js]: 🤣 alright. I'll use the same todo you've left in the `assets.reducer`

File: src/components/SecurityUser/AddSecurityUserComponent.jsx (hunk 2 above)
Comment (Arthur236) [reply src/components/SecurityUser/AddSecurityUserComponent.jsx]: Cool. I'll fix that

**Pull request merged**